### ISSUE TITLE
Redis / Lettuce CacheTransport and Provider

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18149]
dotCMS needs to standardize on an external caching layer. We have hedged our bets between Hazelcast and Redis and ultimately chose to include Hazelcast as it provided most functionality out of the box, including a pub/sub channel that we could use to invalidate caches between dotCMS instances.

Redis 5.0 provides a data type called Streams which is well suited to act as a lossless pub/sub queue, needed for our cache transport layer in dotCMS.  You can read more about it here:  https://redis.io/topics/streams-intro.

The redis java library `lettuce` https://github.com/lettuce-io/lettuce-core allows us to and subscribe to these streams in redis and use them to provide a log of cache invalidations.  The "log" like nature of Redis streams will allow a dotCMS instance to receive invalidations asynchronously, that were sent when it is not in a state to do so, for example when a node is being cycled or during a period of high GC activity.

We need:

1. `RedisPubSubImpl`
2. `RedisCacheTransportImpl` that uses the `RedisPubSubImpl`
3. `RedisCacheProvider` that uses Lettuce and allows us to use Redis as a CacheProvider

**Comment by wezell** [comment]
Note to QA: Once developed, we need to test the RedisCacheProvider by setting it as the only cacheprovider to be used by dotCMS (except for VelocityMacros) and then smoke testing different functionalities.

**Comment by erickgonzalez** [comment]
https://authoring.dotcms.com/devwiki/redis

**Comment by erickgonzalez** [comment]
There was an issue with the redis stats that weren't loading, fixed here: https://github.com/dotCMS/core/pull/21066

**Comment by bryanboza** [comment]
I will pass it for now, this is an epic,  the QA of this card was moved to a separate card #21100

**Comment by wezell** [comment]
waiting on #21100
